### PR TITLE
UIEH-500: Remove all code related to the old unused 'vendors' endpoint

### PR DESCRIPTION
**Pull request by keathleydavidj**
Resolves: [UIEH-500](https://issues.folio.org/browse/UIEH-500)

## Purpose
Some time ago mod-kb-ebsco's `/vendors` endpoint was phased out in favor of a `/providers` endpoint for better semantics. This PR is meant to remove all of the now unused code left over from that name change, reducing old cruft and module size.

### CODE REVIEW

File: spec/requests/packages_spec.rb [hunk 1]
```diff
@@ -460,33 +460,6 @@
     end
   end
 
-  describe 'getting a package with included vendor' do
```
Comment (keathleydavidj) [inline spec/requests/packages_spec.rb:463]: These were the only places I had any uncertainty about. I'm not sure if these tests needed to be removed or if they should have been changed to test for `?include=provider` instead.

File: spec/requests/resources_spec.rb [hunk 2]
```diff
@@ -133,33 +133,6 @@
     end
   end
 
-  describe 'getting a resource with included vendor' do
```
Comment (keathleydavidj) [inline spec/requests/resources_spec.rb:136]: Same as above.

Comment (sduvvuri-ebsco) [reply spec/requests/resources_spec.rb:136]: @keathleydavidj : Please check if there are similar tests for provider. For Example: `getting a package with included provider` and `getting a resource with included provider`. If there are, then these can safely be deleted. Else, please adjust them to look for `provider` instead. Hope this helps. Other than verifying the open questions you have on this PR, the changes look good to me!

**Review by carolegodfrey — APPROVED** [review]
LGTM!

**Pull request merged**